assets/app/scripts/services/logLinks.js

```diff
@@ -5,34 +5,39 @@ angular.module('openshiftConsole')
     '$anchorScroll',
     '$document',
     '$location',
-    '$timeout',
     '$window',
-    function($anchorScroll, $document, $location, $timeout, $window) {
+    'BREAKPOINTS',
+    function($anchorScroll, $document, $location, $window, BREAKPOINTS) {
       // TODO (bpeterse): a lot of these functions are generic and could be moved/renamed to
       // a navigation oriented service.
 
-      var breakpoint = {
-        'mobile': 768
-      };
 
-      var scrollTop = function() {
-        if($(window).width() < breakpoint.mobile) {
-          $window.scrollTo(null, 0);
+      var scrollTop = function(node, isWindow) {
+        if(isWindow) {
+          node.scrollTo(null, 0);
         } else {
-          var objDiv = $(".middle-container.has-scroll")[0];
-          objDiv.scrollTop = 0;
+          node.scrollTop = 0;
         }
       };
 
-      var scrollBottom = function() {
-        if($(window).width() < breakpoint.mobile) {
-           $window.scrollTo(null, $(document).height() - $(window).height());
+
+      // $window and $document do not quite provide the same API as jQuery's wrapper, unfortunately
+      // this is sad face for testability :(
+      // caching these selectors once since fns below may be called many times per second
+      var $doc = $(document);
+      var $win = $(window);
+
+      var scrollBottom = function(node, isWindow) {
```

Comment:
Rather than passing `isWindow`, I'd rather just pass `null` for node.

```
var scrollBottom = function(node) {
    if(!node) {
        $window.scrollTo(null, $doc.height() - $win.height());
    } else {
        node.scrollTop = node.scrollHeight;
    }
}
```